src/waitforit.js

```diff
@@ -0,0 +1,90 @@
+const scopedHandlers = new Map();
+const scopedOnceHandlers = new Map();
+const scopedPayloads = new Map();
+
+function ensureScope(scope) {
+  if (scopedHandlers.has(scope)) {
+    return;
+  }
+
+  scopedHandlers.set(scope, {});
+  scopedOnceHandlers.set(scope, {});
+  scopedPayloads.set(scope, {});
+}
+
+export function destroyScope(scope) {
+  scopedHandlers.delete(scope);
+  scopedOnceHandlers.delete(scope);
+  scopedPayloads.delete(scope);
+}
+
+export function waitFor(scope, it, handler) {
+  ensureScope(scope);
+
+  const handlers = scopedHandlers.get(scope);
+  const payloads = scopedPayloads.get(scope);
```

Comment:
`scopedPayloads` only ever is set in `ensureScope()` and then only with the value `{}`. Is that the only value that a scope payload should have?

Reply:
Ah, I found where it is set. I'm not sure I understand its purpose though.

Reply:
Is the purpose `scopedPayloads` so that late bound handlers get the last value that was published for that event?

Reply:
Yep